src/lib/Microsoft.Health.Fhir.Ingest/Service/CachedResourceIdentityService.cs

```diff
@@ -70,5 +70,7 @@ protected virtual void Dispose(bool disposing)
         protected abstract Task<IDictionary<ResourceType, string>> ResolveResourceIdentitiesInternalAsync(IMeasurementGroup input);
 
         protected abstract string GetCacheKey(IMeasurementGroup input);
+
+        public abstract ResourceIdentityServiceType GetResourceIdentityServiceType();

```

Comment:
>GetResourceIdentityServiceType [](http://example.com/codeflow?start=52&length=30)

Can you add some details on how this is used and why it is needed?

Reply:
Thanks for your feedback on this. My intention was to improve the code by injecting the IdentityServices in ProcessorStartup. However I agree that with these changes, we're creating all possible services (Create, Lookup, and LookupWithEncounter) and then selecting a match. I've gone ahead and reverted the changes here